api/client/client.go

```diff
@@ -2170,3 +2173,79 @@ func (c *Client) GenerateCertAuthorityCRL(ctx context.Context, req *proto.CertAu
 	resp, err := c.grpc.GenerateCertAuthorityCRL(ctx, req)
 	return resp, trail.FromGRPC(err)
 }
+
+// ListResources returns a paginated list of nodes that the user has access to.
+// nextKey can be used as startKey in another call to ListResources to retrieve
+// the next page.
+// It will return a `trace.LimitExceeded` error if the page exceeds gRPC max
+// message size.
+func (c *Client) ListResources(ctx context.Context, req proto.ListResourcesRequest) ([]types.Resource, string, error) {
```

Comment:
Nit: Use named returned arguments to indicate what they mean.
```suggestion
func (c *Client) ListResources(ctx context.Context, req proto.ListResourcesRequest) (resources []types.Resource, nextKey string, err error) {
```